packages/plugin-core/src/commands/PublishDevCommand.ts

```diff
@@ -0,0 +1,75 @@
+import { DENDRON_COMMANDS } from "../constants";
+import {
+  getSiteRootDirPath,
+  checkPreReq,
+  NextJSPublishUtils,
+} from "../utils/site";
+import { BasicCommand } from "./base";
+import fs from "fs-extra";
+import _ from "lodash";
+import { window } from "vscode";
+import { DENDRON_EMOJIS } from "@dendronhq/common-all";
+
+type CommandOutput = {
+  pid: number;
+};
+
+export class PublishDevCommand extends BasicCommand<CommandOutput> {
+  key = DENDRON_COMMANDS.PUBLISH_DEV.key;
+
+  async gatherInputs(): Promise<any> {
+    return {};
+  }
+
+  async sanityCheck() {
+    const sitePath = getSiteRootDirPath();
+    if (!fs.existsSync(sitePath)) {
+      fs.ensureDirSync(sitePath);
+    }
+    return checkPreReq();
+  }
+
+  async execute() {
+    const ctx = "PublishDevCommand";
+    this.L.info({ ctx, msg: "enter" });
+
+    const prepareOut = await NextJSPublishUtils.prepareNextJSExportPod();
+    const { enrichedOpts, wsRoot, cmd, nextPath } = prepareOut;
+    this.L.info({ ctx, msg: "prepare", enrichedOpts, nextPath });
+
+    if (_.isUndefined(enrichedOpts)) {
+      return {};
+    }
+
+    const isInitialized = await NextJSPublishUtils.isInitialized(wsRoot);
+    if (!isInitialized) {
+      await NextJSPublishUtils.initialize(nextPath);
```

Comment:
can we add `vscode.withProgress` bars around here?

Reply:
It is already wrapped around `withProgress` where this is implemented

Reply:
got it. i missed that. its all good then